src/Microsoft.SqlTools.Hosting.v2/Extensibility/ExtensionServiceProvider.cs

```diff
@@ -83,8 +86,23 @@ public static ExtensionServiceProvider CreateFromAssembliesInDirectory(string di
         public static ExtensionServiceProvider Create(IEnumerable<Assembly> assemblies)
         {
             Validate.IsNotNull(nameof(assemblies), assemblies);
+            foreach(var assembly in assemblies)
+            {
+                assemblyCache.AddOrUpdate(assembly.FullName, assembly, (_, previous) => {
+                    return assembly;
+                });
+            }
             return new ExtensionServiceProvider(conventions => new ContainerConfiguration().WithAssemblies(assemblies, conventions));
         }
+        
+        public static ExtensionServiceProvider AddAssembly(string path)
+        {            
```

Comment:
Would prefer to see this be

        public static ExtensionServiceProvider CreateFromAssemblies(IEnumerable<string> assemblyPaths) {...} if you're going to be adding a new method. You could also just convert the assembly path to an Assembly in the calling code and then just call the existing Create